library/src/main/java/com/digi/xbee/api/AbstractXBeeDevice.java

```diff
@@ -3280,6 +3290,62 @@ private boolean enterATCommandMode() throws InvalidOperatingModeException, Timeo
 		return false;
 	}
 	
+	/**
+	 * Exits AT command mode. The XBee device has to be in command mode.
+	 * 
+	 * @throws InterfaceNotOpenException if this device connection is not open.
+	 */
+	private void exitATCommandMode() {
+		if (!connectionInterface.isOpen())
+			throw new InterfaceNotOpenException();
+		
+		try {
+			// Send the exit ('ATCN\r') command.
+			connectionInterface.writeData(COMMAND_MODE_EXIT.getBytes());
+			// Wait the guard time.
+			Thread.sleep(DEFAULT_GUARD_TIME);
+		} catch (IOException e) {
+			logger.error(e.getMessage(), e);
+		} catch (InterruptedException e) {
+			logger.error(e.getMessage(), e);
+		}
+	}
+	
+	/**
+	 * Gets and returns the actual operating mode of the XBee device reading
+	 * the {@code AP} parameter in AT command mode.
+	 * 
+	 * @return The actual operating mode of the XBee device or
+	 *         {@code OperatingMode.UNKNOWN} if the mode could not be read.
+	 * 
+	 * @throws InterfaceNotOpenException if this device connection is not open.
+	 */
+	private OperatingMode getActualMode() throws InvalidOperatingModeException {
+		if (!connectionInterface.isOpen())
+			throw new InterfaceNotOpenException();
+		
+		byte[] readData = new byte[256];
+		try {
+			// Clear the serial input stream.
+			if (connectionInterface instanceof AbstractSerialPort)
+				((AbstractSerialPort)connectionInterface).purge();
+			// Send the 'AP' command.
+			connectionInterface.writeData(COMMAND_MODE_AP.getBytes());
+			Thread.sleep(100);
+			// Read the 'AP' answer.
+			int readBytes = connectionInterface.readData(readData);
+			if (readBytes == 0)
+				return OperatingMode.UNKNOWN;
+			// Return the corresponding operating mode for the AP answer.
+			return OperatingMode.get(Integer.parseInt(new String(readData).trim(), 16));
```

Comment:
The `parseInt` method could throw a `NumberFormatException` if the response is not a number, should we catch that just in case?

Reply:
I'll add the exception to the existing catch clause.